test/helpers/cursor.eachAsync.test.js

```diff
@@ -134,4 +134,61 @@ describe('eachAsync()', function() {
 
     return eachAsync(next, fn, { batchSize, parallel });
   });
+
+  it('executes all documents and aggregates errors if continueOnError set (gh-6355)', () => {
+    const max = 3;
+    let numCalled = 0;
+    function next(cb) {
+      setImmediate(() => {
+        if (++numCalled > max) {
+          return cb(null, null);
+        }
+        cb(null, { num: numCalled });
+      });
+    }
+
+    function fn(doc) {
+      return doc.num % 2 === 1 ? Promise.reject(new Error(`Doc number ${doc.num}`)) : null;
+    }
+
+    return eachAsync(next, fn, { continueOnError: true }).
+      then(() => { throw new Error('Expected error'); },
+        err => {
+          assert.ok(err);
+          assert.equal(err.name, 'EachAsyncMultiError');
+          assert.equal(err.errors.length, 2);
+          assert.equal(err.errors[0].message, 'Doc number 1');
+          assert.equal(err.errors[1].message, 'Doc number 3');
+        });
```

Comment:
I learned from [a useful guide](http://thecodebarbarian.com/async-functions-in-javascript.html#:~:text=Another%20common%20pattern%20is%20to%20use%20Promise%23catch().) a better way to handle errors 😄
```suggestion
    const err = await eachAsync(next, fn, { continueOnError: true }).then(()=> null, err=> err);
    assert.ok(err);
    assert.equal(err.name, 'EachAsyncMultiError');
    assert.equal(err.errors.length, 2);
    assert.equal(err.errors[0].message, 'Doc number 1');
    assert.equal(err.errors[1].message, 'Doc number 3');
```